### ISSUE TITLE
Does not work with update 0.54E

### ISSUE THREAD

**Issue by FrankBKK** [Wehmeyer100/CoI.Mod.Better#45]
The game failed to initialize likely due to a mod, see logs for more info.

Mafi.Core.Game.GameBuilderException: An exception was thrown while registering mod 'BetterMod' (CoI.Mod.Better.BetterMod) ---> Mafi.Core.Mods.DataRegistrationException: MissingMethodException thrown on registration of data 'CoI.Mod.Better.Buildings.BigStorages'. ---> System.MissingMethodException: Method not found: void .Gfx..ctor(string,Mafi.RelTile3f,Mafi.Option`1<string>,Mafi.ColorRgba,bool,System.Nullable`1<Mafi.Core.Entities.Static.Layout.LayoutEntityProto/VisualizedLayers>,System.Nullable`1<Mafi.Collections.ImmutableCollections.ImmutableArray`1<Mafi.Core.Entities.Static.Layout.ToolbarCategoryProto>>,bool,int)
  at CoI.Mod.Better.Buildings.BigStorages.RegisterData (Mafi.Core.Mods.ProtoRegistrator registrator) [0x00080] in F:\Prgramming\Github\CoI.Mod.Better\Source Code\Buildings\BigStorages.cs:45 
  at Mafi.Core.Mods.ProtoRegistrator.RegisterData (Mafi.Core.Mods.IModData dataClass) [0x00034]  
   --- End of inner exception stack trace ---
  at Mafi.Core.Mods.ProtoRegistrator.RegisterData (Mafi.Core.Mods.IModData dataClass) [0x0005f]  
  at Mafi.Core.Mods.ProtoRegistrator.RegisterData[TData] (System.Action`1[T] config) [0x00010]  
  at CoI.Mod.Better.BetterMod.RegisterPrototypes (Mafi.Core.Mods.ProtoRegistrator registrator) [0x00054] in F:\Prgramming\Github\CoI.Mod.Better\Source Code\BetterMod.cs:69 
  at Mafi.Core.Game.GameBuilder.RegisterModsPrototypesOrThrow (Mafi.Collections.ImmutableCollections.ImmutableArray`1[T] mods, Mafi.Core.Mods.ProtoRegistrator protoRegistrator) [0x0001e]  
   --- End of inner exception stack trace ---
  at Mafi.Core.Game.GameBuilder.RegisterModsPrototypesOrThrow (Mafi.Collections.ImmutableCollections.ImmutableArray`1[T] mods, Mafi.Core.Mods.ProtoRegistrator protoRegistrator) [0x00044]  
  at Mafi.Core.Game.GameBuilder.registerProtos (Mafi.Collections.ImmutableCollections.ImmutableArray`1[T] mods, Mafi.Collections.ImmutableCollections.ImmutableArray`1[T] configs, System.Action`2[T1,T2] protosDbRegFnBeforeMods, System.Action`2[T1,T2] protosDbRegFnAfterMods) [0x0001a]  
  at Mafi.Core.Game.GameBuilder+<BuildLoadedGameTimeSliced>d__2.MoveNext () [0x0014d]  
  at Mafi.Unity.GamePlayScene+<loadGame>d__16.MoveNext () [0x001f9]  
  at Mafi.Unity.GamePlayScene+<Initialize>d__14.MoveNext () [0x000a5]  
  at Mafi.Unity.Main.stepSceneInitialization () [0x00000]

**Comment by djdomi** [comment]
I agree to this. Have the same issue - please update us. We will like your mod. keep on! ;-)

#44 #43 #42 is the same like here

**Comment by Creys21** [comment]
Log from  0.5.4g Patch 
The game failed to initialize likely due to a mod, see logs for more info.

Mafi.Core.Game.GameBuilderException: An exception was thrown while registering mod 'BetterMod' (CoI.Mod.Better.BetterMod) ---> Mafi.Core.Mods.DataRegistrationException: MissingMethodException thrown on registration of data 'CoI.Mod.Better.Buildings.BigStorages'. ---> System.MissingMethodException: Method not found: void .Gfx..ctor(string,Mafi.RelTile3f,Mafi.Option`1<string>,Mafi.ColorRgba,bool,System.Nullable`1<Mafi.Core.Entities.Static.Layout.LayoutEntityProto/VisualizedLayers>,System.Nullable`1<Mafi.Collections.ImmutableCollections.ImmutableArray`1<Mafi.Core.Entities.Static.Layout.ToolbarCategoryProto>>,bool,int)
  at CoI.Mod.Better.Buildings.BigStorages.RegisterData (Mafi.Core.Mods.ProtoRegistrator registrator) [0x00080] in F:\Prgramming\Github\CoI.Mod.Better\Source Code\Buildings\BigStorages.cs:45 
  at Mafi.Core.Mods.ProtoRegistrator.RegisterData (Mafi.Core.Mods.IModData dataClass) [0x00034]  
   --- End of inner exception stack trace ---
  at Mafi.Core.Mods.ProtoRegistrator.RegisterData (Mafi.Core.Mods.IModData dataClass) [0x0005f]  
  at Mafi.Core.Mods.ProtoRegistrator.RegisterData[TData] (System.Action`1[T] config) [0x00010]  
  at CoI.Mod.Better.BetterMod.RegisterPrototypes (Mafi.Core.Mods.ProtoRegistrator registrator) [0x00054] in F:\Prgramming\Github\CoI.Mod.Better\Source Code\BetterMod.cs:69 
  at Mafi.Core.Game.GameBuilder.RegisterModsPrototypesOrThrow (Mafi.Collections.ImmutableCollections.ImmutableArray`1[T] mods, Mafi.Core.Mods.ProtoRegistrator protoRegistrator) [0x0001e]  
   --- End of inner exception stack trace ---
  at Mafi.Core.Game.GameBuilder.RegisterModsPrototypesOrThrow (Mafi.Collections.ImmutableCollections.ImmutableArray`1[T] mods, Mafi.Core.Mods.ProtoRegistrator protoRegistrator) [0x00044]  
  at Mafi.Core.Game.GameBuilder.registerProtos (Mafi.Collections.ImmutableCollections.ImmutableArray`1[T] mods, Mafi.Collections.ImmutableCollections.ImmutableArray`1[T] configs, System.Action`2[T1,T2] protosDbRegFnBeforeMods, System.Action`2[T1,T2] protosDbRegFnAfterMods) [0x0001a]  
  at Mafi.Core.Game.GameBuilder+<BuildLoadedGameTimeSliced>d__2.MoveNext () [0x0014d]  
  at Mafi.Unity.GamePlayScene+<loadGame>d__16.MoveNext () [0x001f9]  
  at Mafi.Unity.GamePlayScene+<Initialize>d__14.MoveNext () [0x000a5]  
  at Mafi.Unity.Main.stepSceneInitialization () [0x00000]

**Comment by Wehmeyer100** [comment]
hey, 

I am already working on an update, however there are some changes that I have to painstakingly change one by one. This takes a little time, because I unfortunately have little time privately.

I'll get back to you when I'm done.

Greetings,
Wehmeyer

**Comment by Regzzzm** [comment]
Thank you for your time Wehmeyer. it is greatly appreciated.

**Comment by TheDevilYouKnow420** [comment]
Hello there! Someone came out with a temp fix that works as a stop gap untill the real one can be updated! https://github.com/Darkvoid090/CoI.Mod.Better.tempfix

LOVE the mod by the way, please keep the amazing work up my friend!

**Comment by djdomi** [comment]
why don't you push it as a merge request?
